### PR TITLE
masternodelist "extra" command missing filter functionality

### PR DESCRIPTION
**Pull request by TimothyBanks**
Added filters on the status and outpoint when executing the "extra' command on the masternodelist function.

### CODE REVIEW

**Review by a-ok123 — DISMISSED** [review]

File: src/mnode/rpc/masternode.cpp [hunk 1]
```diff
@@ -115,6 +115,10 @@ Available modes:
         masterNodeCtrl.masternodeManager.UpdateLastPaid(pindex);
     }
 
+    auto is_filtered = [&](const auto&... filters) -> bool {
```
Comment (akobrin1) [inline src/mnode/rpc/masternode.cpp:118]: As I understand, the meaning of this lambda is the opposite: is_not_filtered - i.e. to check that strFilter is not in filters. If strFilter is defined - only items that pass this filter (that have substring) will be added to the output.

Comment (akobrin1) [reply src/mnode/rpc/masternode.cpp:118]: also can make this lambda const

File: src/mnode/rpc/masternode.cpp [hunk 2]
```diff
@@ -146,16 +150,15 @@ Available modes:
             {
                 case RPC_CMD_MNLIST::activeseconds:
                 {
-                    if (!strFilter.empty() && strOutpoint.find(strFilter) == string::npos)
+                    if (is_filtered(strOutpoint))
                         continue;
                     obj.pushKV(strOutpoint, (int64_t)(mn.lastPing.sigTime - mn.sigTime));
```
Comment (akobrin1) [inline src/mnode/rpc/masternode.cpp:155]: BTW, can use obj.pushKV(move(strOutpoint), ....) in all cases

**Review by akobrin1 — CHANGES_REQUESTED** [review]

File: src/mnode/rpc/masternode.cpp (hunk 1 above)
Comment (TimothyBanks) [reply src/mnode/rpc/masternode.cpp:118]: The context that the lambda is being used is asking the question if the current item is filtered and if so, continuing to the next item.  I could change the conditionals to be something like `if (!is_not_filter(...)) continue;` but that's a double negative.

Sure, I can make it const.

File: src/mnode/rpc/masternode.cpp (hunk 2 above)
Comment (TimothyBanks) [reply src/mnode/rpc/masternode.cpp:155]: Ok.  That seems outside the scope of this issue, but I can make that change.

File: src/mnode/rpc/masternode.cpp (hunk 1 above)
Comment (TimothyBanks) [reply src/mnode/rpc/masternode.cpp:118]: I think the parameter name is what causes confusion.  `strFilter` is really a search term not a filter term.  We are using that variable to find the rows we want as opposed to the ros we don't want.

**Comment by TimothyBanks** [comment]
Updated.

**Review by akobrin1 — CHANGES_REQUESTED** [review]
Please rebase this branch on master. It may have some conflicts

**Comment by TimothyBanks** [comment]
Do a rebase commit or a merge commit?

**Comment by a-ok123** [comment]
We merge them

**Pull request closed without merging**